### PR TITLE
feat: Reset the password

### PR DESCRIPTION
**Pull request by YvesLongchamp**
Add the reset password feature, with 3 new routes : 

- GET `users/forgot`
- GET `/users/:userID/reset/:token`
- POST `/users/:userID/reset/:token`

### CODE REVIEW

File: src/bl/requester.ts [hunk 1]
```diff
@@ -74,8 +75,10 @@ export class RequesterFactory {
 
     static async fromToken(token: RequesterToken): Promise<Requester> {
         try {
-            const tokenData = await RequesterFactory.retrieveTokenData(token)
-            const user = await UserFacade.getById(tokenData.user.uid)
+            // const tokenData = await RequesterFactory.retrieveTokenData(token)
+            // console.log('tokenData: ' + tokenData)
```
Comment (Porquepix) [inline src/bl/requester.ts]: ???

File: src/bl/userFacade.ts [hunk 2]
```diff
@@ -98,4 +110,55 @@ export class UserFacade {
...
+    static async reset(email: string): Promise<User> {
+        if (!email) {
+            throw new BadRequest('This user does not exist')
+        }
+        try {
+            const user = await UserFacade.getByEmail(email)
+            const token = uuid('photon.igpolytech.fr', uuid.DNS)
+            user.resetToken = token
+            user.resetTimeStamp = new Date()
+            console.log(user.resetTimeStamp)
```
Comment (Porquepix) [inline src/bl/userFacade.ts]: ???

File: src/bl/userFacade.ts [hunk 3]
```diff
@@ -98,4 +110,55 @@ export class UserFacade {
...
+
+            sendMail(email, resetPassword(user.username, user.id, token))
+            
+            return await getRepository(User).save(user)
```
Comment (Porquepix) [inline src/bl/userFacade.ts]: Maybe save the user first then send the mail.

File: src/entities/user.ts [hunk 4]
```diff
@@ -104,6 +104,18 @@ export class User {
     })
     confirmationToken: string | null
 
+    @Column({
+        type: 'varchar',
+        nullable: true
+    })
+    resetToken: string | null
```
Comment (Porquepix) [inline src/entities/user.ts]: can you add `select: false` for token (email and password) and timestamp. See the password field.

**Review by Porquepix — CHANGES_REQUESTED** [review]

File: src/bl/requester.ts (hunk 1 above)
Comment (YvesLongchamp) [reply src/bl/requester.ts]: oops, was for the notifications ^^'

File: src/bl/userFacade.ts (hunk 3 above)
Comment (YvesLongchamp) [reply src/bl/userFacade.ts]: ok

File: src/bl/userFacade.ts (hunk 2 above)
Comment (YvesLongchamp) [reply src/bl/userFacade.ts]: It was for the test, the `console.log()`. I will remove it.

File: src/entities/user.ts (hunk 4 above)
Comment (YvesLongchamp) [reply src/entities/user.ts]: Ok, but you will have to explain why ;)

**Review by Porquepix — APPROVED** [review]

**Pull request merged**